packages/server-side-render/src/server-side-render.js

```diff
@@ -83,17 +83,17 @@ export class ServerSideRender extends Component {
 
 	render() {
 		const response = this.state.response;
-		const { className } = this.props;
+		const { className, onEmptyResponse, onErrorResponse, onNullResponse } = this.props;
 		if ( response === '' ) {
-			return (
+			return !! onEmptyResponse ? onEmptyResponse( this ) : (
 				<Placeholder
 					className={ className }
 				>
 					{ __( 'Block rendered as empty.' ) }
 				</Placeholder>
 			);
 		} else if ( ! response ) {
-			return (
+			return !! onNullResponse ? onNullResponse( this ) : (
```

Comment:
What is the relevant use case for a null response? Can onEmptyResponse handle all falsy values?

Reply:
Looking at the code, it seems the `! response` is ran when response is null. This seems to happen when things have not yet loaded? It shows a spinner. I was thinking by allowing each of these to have their own callback you could render a custom loading state too.

Does this need revising @jorgefilipecosta ?

Reply:
Ya it looks like null is used to indicate a "loading" state. So there's value in leaving it as is.  However, I would suggest a slight name change here to make the public exposed api a bit clearer.  So instead of `NullResponse` for the render prop (continuing my earlier comment for the review), maybe make it `LoadingResponsePlaceholder`.